syntax = "proto3";

import "dog.proto";

package StackService;



message Empty {}

message StackBoolResponse {
    bool value = 1;
}

message StackSizeResponse {
    int32 value = 1;
}


service StackServer {

    rpc Push(DogType.Dog) returns (Empty) {}

    rpc Pop(Empty) returns (DogType.Dog) {}

    rpc Size(Empty) returns (StackSizeResponse) {}

    rpc IsEmpty(Empty) returns (StackBoolResponse) {}

    rpc Top(Empty) returns (DogType.Dog) {}

    rpc Clear(Empty) returns (StackBoolResponse) {}
}
